### PR TITLE
add `trivy` to Makefile and build image

### PR DESCRIPTION
**Pull request by trevorwhitney**
`trivy` is a command line utility that can scan our images for vulnerabilities. For example, this is how I checked #10217 addressed the CVEs we were hoping it would

```
❯ trivy i grafana/loki:release-2.8.x-d784431
2023-08-11T12:34:15.272-0600    INFO    Vulnerability scanning is enabled
2023-08-11T12:34:15.272-0600    INFO    Secret scanning is enabled
2023-08-11T12:34:15.272-0600    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-08-11T12:34:15.272-0600    INFO    Please see also https://aquasecurity.github.io/trivy/dev/docs/scanner/secret/#recommendation for faster secret detection
2023-08-11T12:34:15.283-0600    INFO    Detected OS: alpine
2023-08-11T12:34:15.283-0600    INFO    Detecting Alpine vulnerabilities...
2023-08-11T12:34:15.286-0600    INFO    Number of language-specific files: 1
2023-08-11T12:34:15.286-0600    INFO    Detecting gobinary vulnerabilities...

grafana/loki:release-2.8.x-d784431 (alpine 3.18.2)

Total: 6 (UNKNOWN: 0, LOW: 0, MEDIUM: 6, HIGH: 0, CRITICAL: 0)

┌────────────┬───────────────┬──────────┬───────────────────┬───────────────┬─────────────────────────────────────────────────────────────┐
│  Library   │ Vulnerability │ Severity │ Installed Version │ Fixed Version │                            Title                            │
├────────────┼───────────────┼──────────┼───────────────────┼───────────────┼─────────────────────────────────────────────────────────────┤
│ libcrypto3 │ CVE-2023-2975 │ MEDIUM   │ 3.1.1-r1          │ 3.1.1-r2      │ AES-SIV cipher implementation contains a bug that causes it │
│            │               │          │                   │               │ to ignore empty...                                          │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-2975                   │
│            ├───────────────┤          │                   ├───────────────┼─────────────────────────────────────────────────────────────┤
│            │ CVE-2023-3446 │          │                   │ 3.1.1-r3      │ Excessive time spent checking DH keys and parameters        │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-3446                   │
│            ├───────────────┤          │                   ├───────────────┼─────────────────────────────────────────────────────────────┤
│            │ CVE-2023-3817 │          │                   │ 3.1.2-r0      │ Excessive time spent checking DH q parameter value          │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-3817                   │
├────────────┼───────────────┤          │                   ├───────────────┼─────────────────────────────────────────────────────────────┤
│ libssl3    │ CVE-2023-2975 │          │                   │ 3.1.1-r2      │ AES-SIV cipher implementation contains a bug that causes it │
│            │               │          │                   │               │ to ignore empty...                                          │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-2975                   │
│            ├───────────────┤          │                   ├───────────────┼─────────────────────────────────────────────────────────────┤
│            │ CVE-2023-3446 │          │                   │ 3.1.1-r3      │ Excessive time spent checking DH keys and parameters        │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-3446                   │
│            ├───────────────┤          │                   ├───────────────┼─────────────────────────────────────────────────────────────┤
│            │ CVE-2023-3817 │          │                   │ 3.1.2-r0      │ Excessive time spent checking DH q parameter value          │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-3817                   │
└────────────┴───────────────┴──────────┴───────────────────┴───────────────┴─────────────────────────────────────────────────────────────┘
```

This PR adds that command as `Makefile` target, as well as adding the `trivy` binary to our drone build image so we can add it to our pipelines in the future.

### CODE REVIEW

**Review by poyzannur — APPROVED** [review]
thanks so much for incorporating this!

File: .drone/drone.yml [hunk 1]
```diff
@@ -1908,8 +1908,3 @@ get:
   path: infra/data/ci/packages-publish/gpg
 kind: secret
 name: gpg_private_key
----
-kind: signature
-hmac: d87780f1737fccb4d0409e34fc97ee07e1bd3c45c9b25fae517e7dbd21fc7a5f
-
-...
```
Comment (chaudum) [inline .drone/drone.yml:1915]: Signature got removed

**Review by chaudum — CHANGES_REQUESTED** [review]

File: .drone/drone.yml (hunk 1 above)
Comment (trevorwhitney) [reply .drone/drone.yml:1915]: good catch, will fix

**Comment by trevorwhitney** [comment]
@chaudum fixed

**Review by chaudum — APPROVED** [review]

**Pull request merged**